spacesuit/src/value.rs

```diff
@@ -126,10 +126,28 @@ impl Into<Scalar> for SignedInteger {
 }
 
 impl Add for SignedInteger {
-    type Output = SignedInteger;
+    type Output = Option<SignedInteger>;
 
-    fn add(self, rhs: SignedInteger) -> SignedInteger {
-        SignedInteger(self.0 + rhs.0)
+    fn add(self, rhs: SignedInteger) -> Option<SignedInteger> {
+        let s = self.0 + rhs.0;
+        if s <= (u64::max as i128) && s >= -(u64::max as i128) {
+            return Some(SignedInteger(s));
```

Comment:
no need for `return` and a semicolon since the if/else statement is the last one in the function

```suggestion
            Some(SignedInteger(s))
```

Same for the `mul`.